datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -25,9 +25,144 @@ use arrow::datatypes::{
 use datafusion_common::DataFusionError;
 use datafusion_common::Result;
 
-use crate::type_coercion::{is_datetime, is_decimal, is_interval, is_numeric};
+use crate::type_coercion::is_numeric;
 use crate::Operator;
 
+/// The type signature of an instantiation of binary expression
+struct Signature {
```

Comment:
I wonder if we can use a different name for this struture as there is already a `Signature` in https://docs.rs/datafusion-expr/26.0.0/datafusion_expr/struct.Signature.html

Perhaps `OpSignature` or `BinarySignature` ?

Reply:
As this is module private I personally think this is fine, I generally dislike adding additional qualifiers to types, we can always change down the line

Reply:
What I really dislike is overloading names of things that are different -- my feeble mind can't mentally alias / keep the two separate